tests/test_extensions.py

```diff
@@ -82,7 +82,9 @@ def test_iterate_entry_points():
     ext_iter = extensions.iterate_entry_points()
     assert hasattr(ext_iter, "__iter__")
     ext_list = list(ext_iter)
-    assert all([isinstance(e, EntryPoint) for e in ext_list])
+    # isinstance(...) doesn't work reliably as we don't know if setuptools using
```

Comment:
We should no longer be relying on setuptools/pkg_resources for the entry-points, right?

And both library module and test do the same conditional import...

That is very tricky!